PriceCalculater/IDiscountService.cs

```diff
@@ -12,17 +12,26 @@ public class DiscountService : IDiscountService
    { 
         private readonly decimal _universalpercentage;
         private readonly Dictionary<long, decimal> _UpcDiscountList;
+        private Precednce universalPrecedence= Precednce.after, upcPrecedence=Precednce.after;
       public DiscountService(decimal universalPercentage , Dictionary<long, decimal> UpcDiscountList)
         {
             this._universalpercentage = universalPercentage;
             this._UpcDiscountList = UpcDiscountList;
         }
+    public DiscountService(decimal universalPercentage,Precednce universalPrecedence,
+        Dictionary<long, decimal> UpcDiscountList ,Precednce upcPrecedence)
+    {
+        this._universalpercentage = universalPercentage;
+        this._UpcDiscountList = UpcDiscountList;
+        this.universalPrecedence = universalPrecedence;
+        this.upcPrecedence = upcPrecedence; 
+    }
     public List<Discount> GetDiscountPercentage(Product product) {
       var Discounts = new List<Discount>();
         if (_universalpercentage > 0) Discounts.Add
-                (new Discount(_universalpercentage,DiscountType.universal));
+                (new Discount(_universalpercentage,DiscountType.universal,universalPrecedence));
         if (_UpcDiscountList.ContainsKey(product.UPC)) Discounts.Add
-                (new Discount(_UpcDiscountList[product.UPC],DiscountType.upc));
+                (new Discount(_UpcDiscountList[product.UPC],DiscountType.upc,upcPrecedence));
          return Discounts;
     }
     }
```

Comment:
2 in order to add precedence to the main function , we needed to add some 2 arguments in the constuctor . (one 2 param only , other 2 are deafult and one takes 4 ) 
is that still okay ? or like the mock state , we make consumer add Discount